tokenawarebandit.go

```diff
@@ -703,3 +714,77 @@ func (lb *latencyBucket) add(latency time.Duration) {
 	lb.sum += latency
 	lb.count += 1
 }
+
+const cacheLineSize = 64 // for amd64
+
+// bigWriterBucket keeps counts of latency with reduced lock contention.
+type bigWriterBucket struct {
+	cpus []bigWriterPerCPU
+}
+
+func newBigWriterBucket() bigWriterBucket {
+	cores := cpuid.CPU.LogicalCores
```

Comment:
are we sure how it will behave inside a container?

Reply:
It uses CPUID instruction, so should report total number of cores on the machine. I have verified it returns correct results in our Kubernetes cluster.